connectors/source/firecrawl.py

```diff
@@ -0,0 +1,312 @@
+    
+    if not api_key:
+        return {"error": "Firecrawl API key is required. Set FIRECRAWL_API_KEY environment variable."}
+    
+    try:
+        # Initialize the Firecrawl client
+        firecrawl = FirecrawlApp(api_key=api_key)
+        
+        # Check the crawl status
+        result = firecrawl.check_crawl_status(crawl_id)
+        
+        # Return a more user-friendly response
+        status_info = {
+            "id": crawl_id,
+            "status": result.get("status", "unknown"),
+            "completed_urls": result.get("completed", 0),
+            "total_urls": result.get("total", 0),
+        }
+        
+            
+        return status_info
+    except Exception as e:
+        return {"error": f"Error checking crawl status: {str(e)}"}
+
+
+def _upload_directory_to_s3(local_dir: str, s3_uri: str) -> Dict[str, Any]:
+    """Upload a directory to S3.
+    
+    Args:
+        local_dir: Local directory to upload
+        s3_uri: S3 URI to upload to
+        
+    Returns:
+        Dict with upload stats
+    """
+    # Parse the S3 URI to get bucket and prefix
+    if not s3_uri.startswith("s3://"):
+        raise ValueError("S3 URI must start with 's3://'")
+    
+    # Remove s3:// prefix and split by first /
+    uri_parts = s3_uri[5:].split('/', 1)
+    bucket_name = uri_parts[0]
+    prefix = uri_parts[1] if len(uri_parts) > 1 else ""
+    
+    # Initialize boto3 S3 client
+    s3_client = boto3.client(
+            's3',
+            aws_access_key_id=os.environ.get('AWS_KEY'),
+            aws_secret_access_key=os.environ.get('AWS_SECRET')
+        )
+    
+    # Track upload stats
+    stats = {
+        "uploaded_files": 0,
+        "failed_files": 0,
+        "total_bytes": 0
+    }
+    
+    # Walk through the directory
+    for root, _, files in os.walk(local_dir):
+        for file in files:
+            local_path = os.path.join(root, file)
+            
+            # Determine the S3 key (path within the bucket)
+            # Remove the local_dir prefix from the file path to get relative path
+            relative_path = os.path.relpath(local_path, local_dir)
+            
+            # Create the S3 key by joining the prefix with the relative path
+            # Replace backslashes with forward slashes for S3
+            s3_key = os.path.join(prefix, relative_path).replace("\\", "/")
+            
+            try:
+                # Upload the file
+                s3_client.upload_file(local_path, bucket_name, s3_key)
+                
+                # Update stats
+                stats["uploaded_files"] += 1
+                stats["total_bytes"] += os.path.getsize(local_path)
+            except Exception as e:
+                print(f"Error uploading {local_path}: {str(e)}")
+                stats["failed_files"] += 1
+    
+    return stats
+
+
+async def wait_for_crawl_completion(
+    ctx: Context,
+    crawl_id: str,
+    s3_uri: str,
+    poll_interval: int = 30,
```

Comment:
Wondering if it would be reasonable to reduce the default time to a few seconds ... it could be organised depending on the job I believe.